web/src/lib/utils/folder-utils.ts

```diff
@@ -0,0 +1,14 @@
+export function buildFolderTree(paths: string[]): any {
```

Comment:
could we avoid the `any` return type here? maybe we need to create a `Folder` type to be used in place of the `any` or something like that?